proxy_presentation.php

```diff
@@ -86,14 +90,24 @@
         }
         return strlen($header);
     },
-    'CURLOPT_WRITEFUNCTION'     => function ($curl, $body) use ($jstobereplaced) {
-        // Even if we modify the body, we need to return the original length
+    'CURLOPT_WRITEFUNCTION'     => function ($curl, $body) use ($jstobereplaced, $relativepath) {
+        // Even if we modify the body, we need to return the original length.
         $originalbodylength = strlen($body);
 
         if ($jstobereplaced) {
             $body = str_replace("let url = '/presentation/'", "let url = '../../../presentation/'", $body);
         }
 
+        // HACK: Version 2.3+ replaces to ensure it routes all assets and subsequent calls through the proxy appropriately.
+        if (strpos($relativepath, '2.0') === false) {
+            $proxybase = explode($relativepath, $_SERVER['HTTP_HOST'].$_SERVER['REQUEST_URI'])[0];
+            $body = str_replace('src="/', 'src="', $body);
```

Comment:
Can you add some comments above each of these with a 5s why its needed. Makes it easier to maintain, otherwise we just need to trust the voodoo :)